pkg/api/controller.go

```diff
@@ -57,19 +57,29 @@ func NewController(config *config.Config) *Controller {
 	return &controller
 }
 
-func DefaultHeaders() mux.MiddlewareFunc {
+func (c *Controller) CORSHeaders() mux.MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
 			// CORS
-			response.Header().Set("Access-Control-Allow-Origin", "*")
-			response.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS")
+			c.CORSHandler(response, request)
```

Comment:
Why not inline the method below right here? Only caller right?

Reply:
No, it is also called by engine.Use https://github.com/shimish2/zot/blob/ui-browser-auth/pkg/api/controller.go#L133